packages/ckeditor5-ui/src/list/listview.ts

```diff
@@ -195,4 +191,82 @@ export default class ListView extends View<HTMLUListElement> implements Dropdown
 	public focusLast(): void {
 		this._focusCycler.focusLast();
 	}
+
+	/**
+	 * Registers a list item view in the focus tracker.
+	 *
+	 * @param item
+	 * @param index
+	 */
+	private _registerFocusableListItem( item: ListItemView, index?: number ) {
+		this.focusTracker.add( item.element! );
+		this.focusables.add( item, index );
+	}
+
+	/**
+	 * Removes a list item view from the focus tracker.
+	 *
+	 * @param item
+	 */
+	private _deregisterFocusableListItem( item: ListItemView ) {
+		this.focusTracker.remove( item.element! );
+		this.focusables.remove( item );
+	}
+
+	/**
+	 * Gets a callback that will be called when the `items` collection of a {@link module:ui/list/listitemgroupview~ListItemGroupView}
+	 * change.
+	 *
+	 * @param groupView
+	 * @returns
+	 */
+	private _getOnGroupItemsChangeCallback( groupView: ListItemGroupView ): GetCallback<ListItemsChangeEvent> {
+		return ( evt, data ) => {
+			for ( const removed of data.removed ) {
+				this._deregisterFocusableListItem( removed );
+			}
+
+			for ( const added of Array.from( data.added ).reverse() ) {
+				this._registerFocusableListItem( added, this.items.getIndex( groupView ) + data.index );
+			}
+		};
+	}
+
+	/**
+	 * Registers a list item group view (and its children) in the focus tracker.
+	 *
+	 * @param groupView
+	 * @param groupIndex
+	 */
+	private _registerFocusableItemsGroup( groupView: ListItemGroupView, groupIndex?: number ) {
+		Array.from( groupView.items ).forEach( ( child, childIndex ) => {
+			const registeredChildIndex = typeof groupIndex !== 'undefined' ? groupIndex + childIndex : undefined;
```

Comment:
Shouldn't be enough: `const registeredChildIndex = groupdIndex ? groupIndex + childIndex : undefined;`? or even:
`const registeredChildIndex = groupIndex + childIndex || undefined;`

Reply:
Yes, it works in JS. But TS wants to protect us from adding `undefined` to `Number`

Reply:
Hmm... I tested the first proposal in TS playground and looks fine. Anyway, it's not really important